src/ZoneServer/World/Storage/PersonalStorage.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Melia.Shared.Tos.Const;
+using Melia.Zone.World.Actors.Characters;
+using Melia.Zone.World.Items;
+using Melia.Zone.Network;
+using Melia.Zone.Database;
+using Melia.Zone.World.Actors.Characters.Components;
+
+namespace Melia.Zone.World.Storage
+{
+	/// <summary>
+	/// Personal storage of a single character.
+	/// </summary>
+	public class PersonalStorage : Storage
+	{
+		/// <summary>
+		/// Character that owns this personal storage.
+		/// </summary>
+		public Character Owner { get; private set; }
+
+		/// <summary>
+		/// Whether the owner is currently browsing this storage.
+		/// </summary>
+		public bool IsBrowsing { get; private set; }
```

Comment:
Wouldn't this property be required for other types of storages as well? I assume you would do the same `IsBrowsing` checks for team storage, and we'd definitely need it to check whether globally accessible storages are currently being browsed, right?

Reply:
Well, I thought about this one and I could potentially move that property to the parent class but then I imagined if we want to implement, say, shared storages, we would need this "IsBrowsing" to be some sort of list for the current characters browsing it.. I think this is better being left in child class, as the child should actually implement the logic on how users actually 'browse' the storage.

Reply:
Would we ever allow a storage to be accessed by multiple players at once? It feels like there'd be a big potential for bugs and exploits with that. In RO for example they restricted the guild storage access to one player at a time.

Reply:
I'm leaving this as is for now, as this property being in the children allows for more flexible future implementations of other types of storage.